ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java

```diff
@@ -259,8 +259,10 @@ private XAResource getNewXAResource(Xid xid)
 
 				// JBTM-1255 moved stale check back to bottomUpRecovery
 				if (xids.contains(xid)) {
-					// This Xid is going to be recovered by the AtomicAction
-					xids.remove(xid);
+					// This Xid will hopefully be recovered by the AtomicAction
+					// We do not remove it from the map as there is garbage collection in RecoveryXids already
```

Comment:
Okay this makes sense, I see that the garbage collection is in bottomUpRecovery()